byterun/caml/roots.h

```diff
@@ -35,10 +35,48 @@ CAMLextern void caml_do_local_roots (scanning_action, value *, value *,
 CAMLextern void caml_do_local_roots(scanning_action f, char * bottom_of_stack,
                                     uintnat last_retaddr, value * gc_regs,
                                     struct caml__roots_block * local_roots);
-#endif
+#endif /* NATIVE_CODE */
 
 CAMLextern void (*caml_scan_roots_hook) (scanning_action);
 
+
+/* Hooks to profile systhreads. */
+
+#define ST_NEXT_HOOK(thread_id,bottom,retaddr) \
+  if( st_next_hook != NULL ) st_next_hook(thread_id,bottom,retaddr)
+CAMLextern void (*st_next_hook)(value thread_id, char* bottom,
+                                        uintnat retaddr);
+/*
```

Comment:
Even if that is the intended purpose of this patch, I don't think the descriptions of the hooks should mention allocation profiling. You may simply say 'run OCaml code' instead of 'allocate'.
Also, I would replace `corresponding` by `The`.